pkg/daemon/ceph/client/upgrade.go

```diff
@@ -18,46 +18,99 @@ package client
 
 import (
 	"encoding/json"
+	"errors"
 	"fmt"
+	"strings"
+	"time"
 
 	"github.com/rook/rook/pkg/clusterd"
 	cephver "github.com/rook/rook/pkg/operator/ceph/version"
+	"github.com/rook/rook/pkg/util"
 )
 
 // CephDaemonsVersions is a structure that can be used to parsed the output of the 'ceph versions' command
 type CephDaemonsVersions struct {
-	Mon     map[string]int `json:"mon,omitempty"`
-	Osd     map[string]int `json:"osd,omitempty"`
-	Mgr     map[string]int `json:"mgr,omitempty"`
-	Mds     map[string]int `json:"mds,omitempty"`
-	Overall map[string]int `json:"overall,omitempty"`
+	Mon       map[string]int `json:"mon,omitempty"`
+	Mgr       map[string]int `json:"mgr,omitempty"`
+	Osd       map[string]int `json:"osd,omitempty"`
+	Rgw       map[string]int `json:"rgw,omitempty"`
+	Mds       map[string]int `json:"mds,omitempty"`
+	RbdMirror map[string]int `json:"rbd-mirror,omitempty"`
+	Overall   map[string]int `json:"overall,omitempty"`
 }
 
-func getCephMonVersionString(context *clusterd.Context) (string, error) {
-	output, err := context.Executor.ExecuteCommandWithOutput(false, "", "ceph", "version")
+var (
+	// we don't perform any checks on these daemons
+	// they don't have any "ok-to-stop" command implemented
+	daemonNoCheck = []string{"mgr", "rgw", "rbd-mirror", "nfs"}
+)
+
+func getCephMonVersionString(context *clusterd.Context, clusterName string) (string, error) {
+	args := []string{"version"}
+	command, args := FinalizeCephCommandArgs("ceph", args, context.ConfigDir, clusterName)
+
+	output, err := context.Executor.ExecuteCommandWithOutput(false, "", command, args...)
+	if err != nil {
+		return "", fmt.Errorf("failed to run ceph version. %+v", err)
+	}
+	logger.Debug(output)
+
+	return output, nil
+}
+
+func getCephVersionsString(context *clusterd.Context, clusterName string) (string, error) {
```

Comment:
Could we rename this to `getAllCephDaemonVersionsString` to clarify that this gets versions for all daemons? My eyes didn't pick up on the `s` at the end of `Versions`.

Reply:
bump